File: solutions/c/collatz-conjecture/1/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int n) {
	if (n <= 0) {
		return -1;
	} else if (n == 1) {
		return 0;
	} else if (n % 2 == 0) {
		return 1 + steps(n / 2);
	} else {
		return 1 + steps(3 * n + 1);
	}
}

File: solutions/go/collatz-conjecture/1/collatz_conjecture.go
package collatzconjecture

import "errors"

func CollatzConjecture(number int) (int, error) {
	if number < 1 {
		return 0, errors.New("zero or negative numbers not valid input")
	}

	steps := 0

	for number != 1 {
		if number%2 == 0 {
			number = number / 2
		} else {
			number = number*3 + 1
		}
		steps++
	}

	return steps, nil
}
